src/luascript.cpp

```diff
@@ -7714,17 +7713,23 @@ int LuaScriptInterface::luaCreatureAddCondition(lua_State* L)
 
 int LuaScriptInterface::luaCreatureRemoveCondition(lua_State* L)
 {
-	// creature:removeCondition(conditionType[, conditionId = CONDITIONID_COMBAT[, subId = 0[, force = false]]])
+	// creature:removeCondition(conditionType or condition[, conditionId = CONDITIONID_COMBAT[, subId = 0[, force = false]]])
 	Creature* creature = getUserdata<Creature>(L, 1);
 	if (!creature) {
 		lua_pushnil(L);
 		return 1;
 	}
+	
+	Condition* condition = nullptr;
+	if (isUserdata(L, 2)) {
+		condition = getUserdata<Condition>(L, 2);
+	} else {
+		ConditionType_t conditionType = getNumber<ConditionType_t>(L, 2);
+		ConditionId_t conditionId = getNumber<ConditionId_t>(L, 3, CONDITIONID_COMBAT);
+		uint32_t subId = getNumber<uint32_t>(L, 4, 0);
+		condition = creature->getCondition(conditionType, conditionId, subId);
+	}
 
-	ConditionType_t conditionType = getNumber<ConditionType_t>(L, 2);
-	ConditionId_t conditionId = getNumber<ConditionId_t>(L, 3, CONDITIONID_COMBAT);
-	uint32_t subId = getNumber<uint32_t>(L, 4, 0);
-	Condition* condition = creature->getCondition(conditionType, conditionId, subId);
 	if (condition) {
 		bool force = getBoolean(L, 5, false);
```

Comment:
I don't think this will work with a `condition`?

Reply:
Why do you think that? At first I saw it wrong too, but then I realized that if it is possible to pass the condition and eliminate it correctly, since it is verified in the list of conditions of the creature and if it matches in the search then it will eliminate it. 

However if you pass a condition other than the owner then nothing will happen, but who would do that?

Reply:
The force parameter is the only detail, thanks @kornholi for letting us see this detail